### PR TITLE
Modified the master i2c support for the bcm2708 so that i2c repeated

### PR DESCRIPTION
**Pull request by jjackowski**
I changed the i2c master such that repeated starts can be used. It requires polling, so I limited their use to mostly SMBus. Some other non-SMBus cases will also use repeated starts, but only when doing so is allowed by i2c. I tested it with a MLX90614 (an infrared thermometer using SMBus) and a BMP085 (pressure and temperature sensor using I2C) on the same bus.

### CODE REVIEW

File: drivers/i2c/busses/i2c-bcm2708.c [hunk 1]
```diff
@@ -238,7 +239,59 @@ static int bcm2708_i2c_master_xfer(struct i2c_adapter *adap,
...
+		//  - setup data for the interrupt handler in advance
+		//  - poll for it to begin (this is why the special case isn't in the interrupt)
+		//  - setup next transfer
+		//  - handoff to interrupt handler
+		
+		unsigned long bus_hz;
+		u32 cdiv, stat;
+		bus_hz = clk_get_rate(bi->clk);
+		cdiv = bus_hz / baudrate;
+
```
Comment (brabl2) [inline drivers/i2c/busses/i2c-bcm2708.c:258]: This code should be here:
   if (cdiv > 0xffff)
     cdiv = 0xffff;

**Comment by popcornmix** [comment]
Have any users of i2c tested this?
I'm happy to merge it if people who actually use this driver report it works for them.

**Comment by brabl2** [comment]
This is very similar to #318 
May be you could cooperate and create one solution.

**Comment by jjackowski** [comment]
I did make available a kernel module that is a drop-in replacement for Raspbian, and it was downloaded (I have no data on how many times), but I haven't heard much about the results. I did get a report about it not working with a MMA8452Q and python-smbus, but it lacked detail and I lack a MMA8452Q, so I don't know what happened.

**Comment by Ruffio** [comment]
@jjackowski, @popcornmix if this PR is not going to be merged, then there is no point in having it 'hanging around'. Please close it, if it is not going to be merged...

**Comment by Ruffio** [comment]
This PR should be closed as it points to 3.10 and therefore it will never get merged.
It should be resubmitted to 4.4.y if still relevant.

**Pull request closed without merging**